changes/ocd-4373.sql

```diff
@@ -0,0 +1,323 @@
+		into v_standard_criterion_map_cnt
+		from openchpl.standard_criteria_map where standard_id = v_standard.id
+		and certification_criterion_id = v_criterion_id;
+
+		if v_standard_criterion_map_cnt = 0 then
+
+			insert into openchpl.standard_criteria_map(standard_id, certification_criterion_id, last_modified_user)
+			select v_standard.id, v_criterion_id, -1
+			where not exists (
+				select *
+				from openchpl.standard_criteria_map where standard_id = v_standard.id
+				and certification_criterion_id = v_criterion_id);
+
+			raise notice 'Inserted Standard Crtieria Map: % -- % (%)', p_regulatory_text, v_certification_criterion.number, v_criterion_id;
+
+		else
+			raise notice 'Standard Crtieria Map Exists: % -- % (%)', p_regulatory_text, v_certification_criterion.number, v_criterion_id;
+		end if;
+
+		for v_cert_result in select cpd.chpl_product_number, cpd.certification_date, cert_result.*
+		from openchpl.certification_result cert_result
+			inner join openchpl.certified_product_details cpd
+				on cert_result.certified_product_id = cpd.certified_product_id
+		where cert_result.certification_criterion_id = v_criterion_id
+		and cpd.certification_date between v_standard.required_day and coalesce(v_standard.end_day, '2099-12-31')
+		and cert_result.deleted = false
+		loop
+
+			select count(*)
+			into v_cert_result_standard_cnt
+			from openchpl.certification_result_standard
+			where certification_result_id = v_cert_result.certification_result_id
+			and standard_id = v_standard.id;
+
+			if v_cert_result_standard_cnt = 0 then
+				insert into openchpl.certification_result_standard(standard_id, certification_result_id, last_modified_user)
+				select v_standard.id, v_cert_result.certification_result_id, -1
+				where not exists (
+					select *
+					from openchpl.certification_result_standard
+					where certification_result_id = v_cert_result.certification_result_id
+					and standard_id = v_standard.id);
+
+				raise notice 'Inserted Certification Result Standard %  -- %  -- %', v_cert_result.chpl_product_number, v_certification_criterion.number, v_standard.regulatory_text_citation;
+			else
+				raise notice 'Certification Result Standard Exists %  -- %  -- %', v_cert_result.chpl_product_number, v_certification_criterion.number, v_standard.regulatory_text_citation;
+			end if;
+		end loop;
+	end loop;
+end $$;
+
+
+call openchpl.backfill_standards('170.202(a)(2)', 'Direct Project: ONC Applicability Statement for Secure Health Transport, Version 1.2 August 2015', 3, '2016-01-14', '2016-01-14', null, null, 'Aplies to both b1 and b1 cures', array[16, 59, 60, 165]);
+
+call openchpl.backfill_standards('170.202(b)', 'ONC XDR and XDM for Direct Messaging Specification', 3, '2016-01-14', '2016-01-14', null, null, null, array[60]);
+
+call openchpl.backfill_standards('170.202(d)', 'ONC Implementation Guide for Direct Edge Protocols, Version 1.1, June 25, 2014', 3, '2016-01-14', '2016-01-14', null, null, null, array[16, 60, 165]);
```

Comment:
In the spreadsheet where this Standard is applied to b1/Cures, it has two rules listed (2015 and Cures). Does that matter? It's just lost here, because we can't apply a Standard to 2 rules in our data model so I want to confirm that's fine. I actually wonder if a Standard should be applied to a rule at all because it's connected to criteria and whatever rules the associated criteria are connected with are the rules the standard is actually connected with. Now I kind of feel like it's incorrect to link a Standard and a Rule to begin with. If we have to pick one rule are we just picking the earliest one in time? So, we would favor 2015 over Cures? Maybe that's what we're doing here but the spreadsheet says "2015/Cures" which makes it look like the Standard is associated with 2 Rules. Thoughts?

Reply:
The rule does not have anything to do what criteria it is associated with.  It is related to when the Standard was created.  Also, we don't use the data for anything, as of now.